### PR TITLE
Add Azure eventhub connector 

### PR DESCRIPTION
**Pull request by anuruddhal**
## Purpose
- Fixes https://github.com/ballerinax/kubernetes/issues/482

### CODE REVIEW

File: src/azure.eventhub/client.bal [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+        public remote function listConsumerGroups() returns @tainted xml | error {
+            http:Client clientEndpoint = new ("https://" + self.config.resourceUri);
+            http:Request req = new;
+            string token = self.getSASToken();
+            req.addHeader("Authorization", token);
+            var response = clientEndpoint->get("/consumergroups", req);
+            if (response is http:Response) {
+                int statusCode = response.statusCode;
+                io:println("Status code: " + statusCode.toString());
+                io:println(response.getXmlPayload());
```
Comment (kalaiyarasiganeshalingam) [inline src/azure.eventhub/client.bal:288]: Is this needed?

File: src/azure.eventhub/client.bal [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+    # + userProperties - user properties 
+    # + publisherId - publisher ID 
+    # + data - event data
+    # + brokerProperties - broker properties
+    # + partitionId - partition ID
+    # + return - @error if an error occurs
+    public remote function send(string|xml|json|byte[]|io:ReadableByteChannel|mime:Entity[] data,
+        public map<string> userProperties = {}, public map<anydata> brokerProperties = {}, public int partitionId = -1,
+        public string publisherId = "") returns @tainted error? {
+        http:Client clientEndpoint = new ("https://" + self.config.resourceUri);
```
Comment (kalaiyarasiganeshalingam) [inline src/azure.eventhub/client.bal:49]: Do we need to create the client for. every operation?

File: src/azure.eventhub/client.bal [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+    }
+
+    # Send a single event
+    #
+    # + userProperties - user properties 
+    # + publisherId - publisher ID 
+    # + data - event data
+    # + brokerProperties - broker properties
+    # + partitionId - partition ID
+    # + return - @error if an error occurs
```
Comment (hemikak) [inline src/azure.eventhub/client.bal:45]: I think we need to improve the documentation.

File: src/azure.eventhub/client.bal [hunk 4]
```diff
@@ -0,0 +1,378 @@
...
+import ballerina/encoding;
+import ballerina/http;
+import ballerina/io;
+import ballerina/log;
+import ballerina/mime;
+import ballerina/time;
+
+# Description 
+#
+# + config - config Parameter Description
```
Comment (hemikak) [inline src/azure.eventhub/client.bal:27]: Update documentation

File: src/azure.eventhub/client.bal [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+    }
+
+        # Lit available partitions
+        #
+        # + consumerGroupName - consumer group name
+        # + return - Return partition list or error
+        public remote function listPartitions(string consumerGroupName) returns @tainted xml | error {
+            http:Client clientEndpoint = new ("https://" + self.config.resourceUri);
+            http:Request req = new;
+            string token = self.getSASToken();
```
Comment (hemikak) [inline src/azure.eventhub/client.bal:218]: Indentation issue ?

File: src/azure.eventhub/client.bal [hunk 6]
```diff
@@ -0,0 +1,378 @@
...
+            return error(EVENT_HUB_ERROR, message = "error invoking EventHub API ", cause = response);
+        }
+    }
+
+    # Resume a publisher
+    #
+    # + publisherName - publisher name 
+    # + return - Return publisher details or error
+    public remote function resumePublisher(string publisherName) returns @tainted xml|error {
+        http:Client clientEndpoint = new ("https://" + self.config.resourceUri);
```
Comment (hemikak) [inline src/azure.eventhub/client.bal:195]: Can we keep the client at the object body level than creating it in every remote method ?

File: src/azure.eventhub/client.bal [hunk 7]
```diff
@@ -0,0 +1,378 @@
...
+        # Get partition details
+        #
+        # + consumerGroupName - consumer group name
+        # + partitionId - partitionId
+        # + return - Returns partition details
+        public remote function getPartition(string consumerGroupName, int partitionId) returns @tainted xml | error {
+            http:Client clientEndpoint = new ("https://" + self.config.resourceUri);
+            http:Request req = new;
+            string token = self.getSASToken();
+            req.addHeader("Authorization", token);
```
Comment (hemikak) [inline src/azure.eventhub/client.bal:241]: Can we move this to a separate function if it makes sense. It looks duplicated in most remove functions.

**Comment by hemikak** [comment]
Can we remove the empty resources directories ?

**Pull request merged**